deploy_heroku.py

```diff
@@ -43,74 +53,70 @@ def get_request(app: str, api_key: str, endpoint: str, payload: Optional[dict] =
 
     }
 
-    return urllib.request.Request(
+    request = urllib.request.Request(
         f"https://api.heroku.com/apps/{app}/{endpoint}",
         headers=headers,
         data=json.dumps(payload).encode() if payload is not None else None,
     )
 
-
-def get_latest_commit_from_slug(app: str, api_key: str, slug_id: str):
-    request = get_request(app, api_key, Endpoint.SLUGS.format(slug_id))
-
     with urllib.request.urlopen(request) as response:
-        assert response.getcode() == 200, response.getcode()
-        return json.loads(response.read())["commit"]
+        assert response.getcode() in get_status_codes(request.get_method()), response.getcode()
+        return json.loads(response.read())
 
 
 def get_latest_heroku_release(app: str, api_key: str) -> HerokuRelease:
-    request = get_request(app, api_key, Endpoint.RELEASES)
-
-    with urllib.request.urlopen(request) as response:
-        assert response.getcode() in (200, 206), response.getcode()
-        latest_release, *_ = json.loads(response.read())
-        return release_from_response(latest_release)
+    return release_from_response(get_response(app, api_key, Endpoint.RELEASES)[0])
 
 
 def trigger_release_retry(app: str, api_key: str, release: HerokuRelease):
-    request = get_request(
-        app=app,
-        api_key=api_key,
-        endpoint=Endpoint.RELEASES,
-        payload={"slug": release.slug_id, "description": f"Retry of v{release.version}: {release.description}"},
-    )
+    payload = {"slug": release.slug_id, "description": f"Retry of v{release.version}: {release.description}"}
+    return release_from_response(get_response(app, api_key, Endpoint.RELEASES, payload))
 
-    with urllib.request.urlopen(request) as response:
-        assert response.getcode() == 201, response.getcode()
-        return release_from_response(json.loads(response.read()))
+
+def wait_for_release(app, api_key, release_version: int, timeout: int = 300, wait_time: int = 3):
```

Comment:
Finde ich schon super-ärgerlich, dass es sowas in der standard library nicht gibt :( es gibt aber turtle spräche Implementierung! Das ist wichtig!